### PR TITLE
Fix some issue to enable HEVC on iOS.

### PR DESCRIPTION
**Pull request by jianjunz**
- Add some missing classes.
- Define OWT_USE_H265 when rtc_use_h265 is true.

### CODE REVIEW

File: sdk/objc/components/video_codec/RTCCodecSpecificInfoH265.mm [hunk 1]
```diff
@@ -0,0 +1,28 @@
+/*
+ *  Copyright 2017 The WebRTC project authors. All Rights Reserved.
+ *
+ *  Use of this source code is governed by a BSD-style license
+ *  that can be found in the LICENSE file in the root of the source
+ *  tree. An additional intellectual property rights grant can be found
+ *  in the file PATENTS.  All contributing project authors may
+ *  be found in the AUTHORS file in the root of the source tree.
+ */
+ /* This file is borrowed from sdk/objc/components/video_codec/RTCCodecSpecificInfoH265.mm */
```
Comment (taste1981) [inline sdk/objc/components/video_codec/RTCCodecSpecificInfoH265.mm]: borrowed from ...InfoH264.mm

File: sdk/BUILD.gn [hunk 2]
```diff
@@ -553,8 +553,14 @@ if (is_ios || is_mac) {
         "objc/components/video_codec/RTCCodecSpecificInfoH264+Private.h",
         "objc/components/video_codec/RTCCodecSpecificInfoH264.h",
         "objc/components/video_codec/RTCCodecSpecificInfoH264.mm",
+        "objc/components/video_codec/RTCCodecSpecificInfoH265+Private.h",
+        "objc/components/video_codec/RTCCodecSpecificInfoH265.h",
+        "objc/components/video_codec/RTCCodecSpecificInfoH265.mm",
         "objc/components/video_codec/RTCH264ProfileLevelId.h",
         "objc/components/video_codec/RTCH264ProfileLevelId.mm",
+        # TODO: Split this file.
+        "objc/Framework/Headers/WebRTC/RTCVideoCodecH265.h",
```
Comment (taste1981) [inline sdk/BUILD.gn]: should optionally include H265files?

File: sdk/objc/native/src/objc_video_encoder_factory.mm [hunk 3]
```diff
@@ -15,6 +15,7 @@
 #import "base/RTCVideoEncoder.h"
 #import "base/RTCVideoEncoderFactory.h"
 #import "components/video_codec/RTCCodecSpecificInfoH264+Private.h"
+#import "components/video_codec/RTCCodecSpecificInfoH265+Private.h"
```
Comment (taste1981) [inline sdk/objc/native/src/objc_video_encoder_factory.mm:19]: In this file, all 265 related inclusion and call should be protected with OWT_USE_H265 macro .

File: sdk/objc/components/video_codec/RTCCodecSpecificInfoH265.mm (hunk 1 above)
Comment (jianjunz) [reply sdk/objc/components/video_codec/RTCCodecSpecificInfoH265.mm]: Done.

File: sdk/BUILD.gn (hunk 2 above)
Comment (jianjunz) [reply sdk/BUILD.gn]: Done.

File: sdk/objc/native/src/objc_video_encoder_factory.mm (hunk 3 above)
Comment (jianjunz) [reply sdk/objc/native/src/objc_video_encoder_factory.mm:19]: Done.

File: sdk/objc/components/video_codec/RTCCodecSpecificInfoH265.h [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+ *  be found in the AUTHORS file in the root of the source tree.
+ */
+/* This file is borrowed from sdk/objc/components/video_codec/RTCCodecSpecificInfoH264.h. */
+
+#import <Foundation/Foundation.h>
+
+#import "RTCCodecSpecificInfo.h"
+#import "RTCMacros.h"
+
+/** Class for H264 specific config. */
```
Comment (taste1981) [inline sdk/objc/components/video_codec/RTCCodecSpecificInfoH265.h]: H265 specific config

**Review by taste1981 — APPROVED** [review]

File: sdk/objc/components/video_codec/RTCCodecSpecificInfoH265.h (hunk 4 above)
Comment (jianjunz) [reply sdk/objc/components/video_codec/RTCCodecSpecificInfoH265.h]: Done.

**Pull request merged**